state-chain/pallets/cf-environment/src/lib.rs

```diff
@@ -431,13 +433,16 @@ impl<T: Config> Pallet<T> {
 					let params = Self::consolidation_parameters();
 
 					let utxos_to_consolidate =
-						select_utxos_for_consolidation(available_utxos, fee_per_input_utxo, params);
+						select_utxos_for_consolidation(available_utxos, &bitcoin_fee_info, params);
 
 					if utxos_to_consolidate.is_empty() {
 						Err(())
 					} else {
-						let total_fee = utxos_to_consolidate.len() as u64 * fee_per_input_utxo +
-							fee_per_output_utxo + min_fee_required_per_tx;
+						let total_fee = utxos_to_consolidate
```

Comment:
We should put this in a closure and deduplicate it, it's the exact same was above